Opensds_state_machine_design.md

```diff
@@ -0,0 +1,109 @@
+# Opensds state machine design
+
+Status: WIP
+
+Version: Alpha
+
+Author(s): jack li (@lasshpanzer)
+
+## Summary
+
+The State machines of volume,snapshot and attachment show the real-time status and state transitions between actions.
+
+## Motivation
+
+For example, When creating a volume, there is needed to know if it has been created or if something went wrong.
+
+### Goals
+
+Design volume statuses
+
+Design snapshot statuses
+
+Design attachment statuses
+
+### Non-Goals
+
+Some architectures needs to be changed.
```

Comment:
This sentence is confusing.  Can you please clarify?

Also we do need statuses for Group, GroupSnapshot, and Replication later.  Please add them here under Non-Goals for this PR and noted that they will be added later.